Weekend Work for Functions

sczyz opened the pull request:
Created functions, test arrays, and updated package organization

sczyz committed:
add __main__.py

sczyz committed:
Added a test folder

sczyz committed:
First two functions for project

-Implemented functions for importing a csv file and making a column vector out of it in array_import.py.
-Created a file for taking several of those columns and compiling them into a large array (did not start writing said file)
-Created a test .csv file "test1.csv" for testing the array_import.py file

sczyz committed:
Moved test1.csv for organization

sczyz committed:
Reorganized files

Redefined my main python file with matrix composition file

sczyz committed:
Massive sunday workflow

Added several test files, reorganized excessively, attempted to adjust hierarchy

sczyz committed:
Fix import issues

-Moved test_script to main directory
-fixed imports to work for both the test functions and the __main__

sczyz committed:
New function, formatting, comments

Streamlined design and imports in test_script and ml_matrix_composition, defined two new functions in test_functions, added comments in all files describing what each function does

manninosi commented:
Bomstrip.py calls the following on line 24, but the files are not being pointed to correctly. 

    uzzah = open('../tests/test2.csv', "r+b")

Need to add appropriate working directory. Can also use the package "Tkinter" and the function "askopenfilename" to set a variable that points to the specific file. 

array_import.py

Does the "load_2d_coinc_spectrum" look at one of the csv test files? There is a specific "csv" package you can import to read in csv files. Though this probably works just fine. 

Try transferring all of the testing functions from "test_functions.py" to the "test_script.py". That should make it a little easier to run pytest. 

I found that I wasn't able to open the files you had listed in the "form_matrix(n)" function. I think the wrong file divider is being used. Try using "\" instead. However, once I made that change I got an error the I was in an "invalid mode" .

sczyz committed:
Removed bomstrip for now

sczyz committed:
Added .travis.yml

sczyz committed:
Added files for .travis.yml as indicated by class

sczyz committed:
Reorganized files to allow for travis.yml to work

sczyz committed:
Removed ${PYTHON}

sczyz committed:
Fixed importing

sczyz committed:
Fixed datatype of np.arrays to int32

Avoids issues of defining as either int32 or int64

sczyz committed:
testing travis from desktop

sczyz committed:
doc_strings, defining new tests, and testing function parameters

sczyz committed:
last second change to matrix_composition

sczyz committed:
test functions edit

Removed "test_import_sizes" as I determined it to be irrelevant
Added test_file_existence() to test if files actually load if they exist
Added test_different_n_values to determine if the system will load a file regardless of file number

sczyz committed:
changed name of variable

sczyz committed:
Removed test_script.py due to pytest serving this function

sczyz committed:
changed offset in test_functions function test_array_clear() such that it passes

sczyz committed:
Added variance solver for ML and test

Still need to modify the test_variance- is currently accepting an array for A, should be accepting a single value

sczyz committed:
Fixed variance and test_functions to work as intended

Also added two import lines in test_functions to allow for refreshing of variance and ml_matrix_composition files after reloading (lines 12-13). Before, if I changed variance and reran test_variance() it would use the old version. With this added in, it uses the new version. I'm currently not sure why this works this way

manninosi commented:
Looking at bomstrip.py, you shouldn't have to import the packages within the function. That should only occur at the top the of the script. This also could be an artifact of me force merging. 

Looking at at the form_matrix, update the doc string to describe each variable and provide a brief description of what is contained in the file being read-in. I really like how the formatted was established in class, look at my "bit_chop" function for an example of variable description.  

- What does the "huzzah" variable represent?
- What is the difference between "thearr" and "experiemntal_vec". I believe you are going to use one to conduct your ML method? 

For the variance function. I would recommend being a little more explicit in your variable naming.   What is Eqn. 5? I looks like your add an int to the "AS" variable, but then if q is greater than zero it looks like you are trying to access different indices. 

Once you address these I'll go and merge the repositories.

sczyz committed:
Fixed indentation error

sczyz commented:
-bomstrip.py has been removed. I added it to the repo and then I removed it because I realized that it was mostly unnecessary for me right now. So it is an artifact of force-merging.
-I renamed variables to be "inherently commented". I have a nasty habit of, while making a code for the first time, using very poorly descriptive names for things. Huzzah, for example, was just me trying to get it to work, then when it worked, my response (in real life) was "huzzah!" I've since changed the name to a more appropriate one. The array and the vector correspond to the array made from simulation data from each individual radioxenon isotope and the single experimental measurement, respectively.

The equations discussed in the variance function are in the paper that I'm referencing for the ML equations. I'm going to put citation information in the docstring.

sczyz committed:
Variable name changes and docstrings

-Change variable names in ml_matrix_composition to be more inherently clear as to what each variable means
-Changed docstrings to more thoroughly describe variables of functions
-removed unnecessary imports in test_functions

sczyz committed:
Create J and K matrices & associated tests

-Created the functions to make the J and the K matrices for the ML solver
-wrote the tests to test said functions
-composed doc strings for each
-edited variance function to be more robust in the sizing of the D-vector

sczyz committed:
Fixed size of A vector in test_variance in compliance with size of f matrix

manninosi commented:
Great! The functions are more clearly labeled and Travis CI is approving your tests.
